### PR TITLE
feat(auth): add support for file sourced credentials

### PR DESCRIPTION
**Pull request by alvarowolfx**
Fixes [#2608](https://github.com/googleapis/google-cloud-rust/issues/2608)

### CODE REVIEW

File: src/auth/src/credentials/external_account_sources/file_sourced.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            file,
+            format,
+            subject_token_field_name,
+        }
+    }
+}
+
+const JSON_FORMAT_TYPE: &str = "json";
+
+impl SubjectTokenProvider for FileSourcedCredentials {
```
Comment (harkamaljot) [inline src/auth/src/credentials/external_account_sources/file_sourced.rs:49]: Is it possible we can add unit tests here same as other files eg: https://github.com/googleapis/google-cloud-rust/blob/main/src/auth/src/credentials/external_account_sources/url_sourced.rs As right now test coverage tool shows file is missing coverage.

Comment (sai-sunder-s) [reply src/auth/src/credentials/external_account_sources/file_sourced.rs:49]: +1. Please add tests for json type and text type file sources and some negative cases as well.

Comment (alvarowolfx) [reply src/auth/src/credentials/external_account_sources/file_sourced.rs:49]: pushed tests, I was gonna send in another PR, but makes sense to send on this.

File: src/auth/src/credentials/external_account_sources/file_sourced.rs [hunk 2]
```diff
@@ -60,3 +74,92 @@ impl SubjectTokenProvider for FileSourcedCredentials {
         }
     }
 }
+
+#[cfg(test)]
+mod test {
```
Comment (sai-sunder-s) [inline src/auth/src/credentials/external_account_sources/file_sourced.rs:79]: A test with empty file?

**Review by sai-sunder-s — DISMISSED** [review]

File: src/auth/src/credentials/external_account_sources/file_sourced.rs [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+            .unwrap_or(("text".to_string(), String::new()));
+        Self {
+            file,
+            format,
+            subject_token_field_name,
+        }
+    }
+}
+
+const JSON_FORMAT_TYPE: &str = "json";
```
Comment (harkamaljot) [inline src/auth/src/credentials/external_account_sources/file_sourced.rs:47]: should we move this consts file in this module?

Comment (alvarowolfx) [reply src/auth/src/credentials/external_account_sources/file_sourced.rs:47]: will do that in a follow up PR

File: src/auth/src/credentials/external_account_sources/file_sourced.rs (hunk 2 above)
Comment (alvarowolfx) [reply src/auth/src/credentials/external_account_sources/file_sourced.rs:79]: added extra test cases for that

**Review by sai-sunder-s — APPROVED** [review]
